### PR TITLE
Filter markets

### PR DESCRIPTION
**Pull request by anxolin**
This PR adds a new convenient filter that would allow anyone to create a channel, only for the tokens he is interested.

```js
TOKEN_IDS_FILTER=7,4
```

Additionally, there's a lot of documentation that was added to make every config in the example, self explanatory.

### CODE REVIEW

File: src/services/DfusionService.ts [hunk 1]
```diff
@@ -156,10 +167,40 @@ export class DfusionRepoImpl implements DfusionService {
...
+          filter: { buyToken: this._tokenIdsFilter },
+        }),
+      )
+
+      subscriptions.set(
+        'Orders whose Sell Token is ' + tokenListDescription,
+        OrderPlacement({
+          filter: { sellToken: this._tokenIdsFilter },
+        }),
+      )
```
Comment (Velenir) [inline src/services/DfusionService.ts:186]: I thought it should be possible to essentially produce this kind of [A || B topic](https://github.com/ethereumproject/wiki/wiki/JSON-RPC#parameters-58)

And it is, try `await contract.getPastEvents('OrderPlacement', {fromBlock: 10327950, filter: {sellToken:[1,2], buyToken:[3,4]}})` and look at all sell/buy permutations.
It matches everything with everything
```
topics = OrderPlacement && SELL(1 || 2) && BUY(3 || 4)
```



So one subscription of 
```js
OrderPlacement({
          filter: { buyToken: this._tokenIdsFilter, sellToken: this._tokenIdsFilter },
        }),
```
is enough :raised_hands:

Comment (Velenir) [reply src/services/DfusionService.ts:186]:
```suggestion
    if (this._tokenIdsFilter) {
      const tokenListDescription = this._tokenIdsFilter.join(', ')
      subscriptions.set(
        'Orders whose Buy or Sell Token is ' + tokenListDescription,
        OrderPlacement({
          filter: { buyToken: this._tokenIdsFilter, sellToken: this._tokenIdsFilter },
        }),
      )
```

like this

File: .env.example [hunk 2]
```diff
@@ -1,43 +1,149 @@
...
+#  The TCR should implement a basic interface:
+#     function getTokens(uint256 _listId) public view returns (address[] memory) 
+#  One implementation of a TCR is the one done by dxDAO:
+#     https://github.com/nicoelzer/dxDAO-Token-Registry/blob/master/contracts/dxTokenRegistry.sol
 
 # TCR - list of whitelisted tokens
-#  contract address, required
-# Rinkeby:
+# Mainnet: TCR - list of whitelisted tokens
+TCR_CONTRACT_ADDRESS=0x93DB90445B76329e9ed96ECd74e76D8fbf2590d8
```
[the remark this answers is not in the document]
Comment (anxolin) [reply .env.example:88]: Yep, I should have something in the description too.

It makes sense to me. I think probably makes sense to do the same in Mesa. 
Your test TCR was cool, but maybe makes sense to use dxDAO one.

which one we use in PRODUCTION?

[the remark this answers is not in the document]
Comment (alfetopito) [reply .env.example:88]: For telegram, we only use our own TCR, both test and prod.
The config for that is on gitlab though. We have to ask devops to perform the change.

For Mesa, dxdao already uses their own TCR.

File: src/services/DfusionService.ts (hunk 1 above)
Comment (anxolin) [reply src/services/DfusionService.ts:186]: Thanks for this Dima! I'll improve it in another PR. I'll share it already. Promise I'm doing this the moment I get a minute for it :)

Comment (anxolin) [reply src/services/DfusionService.ts:186]: I'll leave it open, and create an issue!

File: .env.example (hunk 2 above)
[the remark this answers is not in the document]
Comment (anxolin) [reply .env.example:88]: I think at this point, probably make sense to use dxDAO one. I think they should have a good criteria for listing.

Unless we want to have full control. I believe that the closer we have our staging to what's used in production the, better.

[the remark this answers is not in the document]
Comment (alfetopito) [reply .env.example:88]: Agree

**Pull request merged**